chabud/model.py

```diff
@@ -40,70 +52,76 @@ def __init__(self, lr: float = 0.001, submission_filepath: str = "submission.csv
         # Save hyperparameters like lr, weight_decay, etc to self.hparams
         # https://pytorch-lightning.readthedocs.io/en/2.0.2/common/lightning_module.html#save-hyperparameters
         self.save_hyperparameters(logger=True)
+        self.model = self._init_model(model_name)
 
-        # First input convolution with same number of groups as input channels
-        self.input_conv = torch.nn.Sequential(
-            torch.nn.Conv2d(
-                in_channels=24, out_channels=64, kernel_size=3, padding=1, groups=4
-            ),
-            torch.nn.SiLU(),
+        # Loss functions
+        self.loss_bce = torch.nn.BCEWithLogitsLoss(
+            pos_weight=torch.tensor(32.0), reduction="mean"
         )
+        # self.loss_dice = DiceLoss(mode="binary", from_logits=True, smooth=0.1)
+        # self.loss_focal = BinaryFocalLoss(alpha=0.25, gamma=2.0)
 
-        # TODO Get backbone architecture from some model zoo
-        # self.backbone = ()
-
-        # Output layers
-        self.output_conv = torch.nn.Conv2d(
-            in_channels=64, out_channels=1, kernel_size=3, padding=1, groups=1
+        # Evaluation metrics to know how good the segmentation results are
+        self.iou = torchmetrics.JaccardIndex(
+            task="binary", threshold=0.5, num_classes=2
         )
 
-        # Loss functions
-        self.loss_bce = torch.nn.BCEWithLogitsLoss(reduction="mean")
-
-        # Evaluation metrics to know how good the segmentation results are
-        self.iou = torchmetrics.JaccardIndex(task="binary", num_classes=2)
+    def _init_model(self, name):
+        if name == "tinycd":
+            return ChangeClassifier(
+                bkbn_name="efficientnet_b4",
+                weights=None,  # not using pretrained weights
```

Comment:
Was getting NaN loss when using the pretrained EfficientNet_B4 weights from torchvision, likely due to the Sentinel-2 image dataset being from a different distribution to ImageNet. Set this to `None` so that we're training from scratch instead.

Reply:
@weiji14 Added `batchnorm` as the first layer of the model - it handles normalizing different bands & works with imagenet pretrained weights.